static/js/public/fsf-language-select.js

```diff
@@ -0,0 +1,58 @@
+// TODO:
+// - add details data for all languages
+// - add button to the docs
+import debounce from '../libs/debounce';
+
+function initFSFLanguageSelect(rootEl) {
+  const flowLinks = [].slice.call(rootEl.querySelectorAll('.p-flow-link'));
+  const flowDetails = [].slice.call(rootEl.querySelectorAll('.p-flow-details'));
+
+  const closeDetails = () => {
+    flowDetails.forEach(e => e.style.display = 'none');
+    flowLinks.forEach(l => l.classList.remove('is-open'));
+  };
+
+  flowLinks.forEach(link => {
+    link.addEventListener('click', (event) => {
+      var link = event.target.closest('.p-flow-link');
+
+      if (link && link.dataset.flowLink) {
+        // find where the next row of links starts to insert details panel before
+        var top = link.getBoundingClientRect().top;
+
+        var nextRow = null;
+        for (var i = flowLinks.indexOf(link); i < flowLinks.length && !nextRow; i++) {
+          if (flowLinks[i].getBoundingClientRect().top > top) {
+            nextRow = flowLinks[i];
+          }
+        }
+        const isOpen = link.classList.contains('is-open');
+
+        closeDetails();
+
+        if (!isOpen) {
+          // find the end of the row of icons to place details panel properly
+          var details = rootEl.querySelector(`[data-flow-details='${link.dataset.flowLink}'`);
+          if (nextRow) {
+            nextRow.parentNode.insertBefore(details, nextRow);
+          } else {
+            rootEl.appendChild(details);
+          }
+          details.style.display = 'block';
+          link.classList.add('is-open');
+        } else {
+          link.classList.remove('is-open');
+        }
+
+        event.preventDefault();
+      }
+    });
+  });
+
+  const onResize = debounce(closeDetails, 500);
```

Comment:
It's because of the positioning and flexbox. When opening details I need to calculate where to inject it between the icons so it appears below the whole flexbox row. When we resize the icons float differently, so it's easier to just close the details and reposition it correctly when it's opened again.